week_1/leap/main.go

```diff
@@ -1,15 +1,16 @@
-// This is a "stub" file. Це є маленький запуск на вашій згоді.
-// It's not a complete solution though; You have to write some code.
-
-// Package leap should have package comment that summarizes what it's about.
-// https://golang.org/doc/effective_go.html#commentary
+// Determine if given year is a leap year
 package leap
 
-// IsLeapYear повинен мати коментар documenting it.
-func IsLeapYear(int) bool {
-	// Тут має бути рішення
-	// написавши код - необхідно запустити тести
-	// Ці коментарі можна видаляти
-	// !ВАЖЛИВО - не забудьте виправити return
-	return false
+// IsLeapYear returns true if its argument is a leap year
+func IsLeapYear(y int) bool {
+	switch {
+	case y%400 == 0:
+		return true
+	case y%100 == 0:
+		return false
+	case y%4 == 0:
+		return true
+	default:
+		return false
+	}
 }
```

Comment:
This solution works, although there is an alternative to solving it with a simple set of logical conditions